feat(aip-136): response message name (#1387)
googleapis/api-linter: docs/rules/0136/response-message-name.md

---
rule:
  aip: 136
  name: [core, '0136', response-message-name]
  summary: Custom methods must have standardized response message names.
permalink: /136/response-message-name
redirect_from:
  - /0136/response-message-name
---

# Custom methods: Response message

> This rule enforces that all custom methods should take a response message
> matching the RPC name, with a `Response` suffix, or the resource being operated
> on [AIP-136][].

Reviewer: ```suggestion
This rule enforces that custom methods have a response message that is named to
match the RPC name with a `Response` suffix, or that is the resource being operated
on, as described in [AIP-136][].
```

Just wanted to tighten it up a bit, take the "should" out of it, the guidance on AIP-136 will dictate the requirement level.

## Details

This rule looks at any method that is not a standard method, and complains if
the name of the corresponding output message does not match the name of the RPC
with the suffix `Response` appended, or the resource being operated on.

> **Note:** To identify the resource being operated on, the rule inspects the
> name path parameter, which maps to the `name` field on the input type checking
> that the resource message derived from `(google.api.resource_reference).type`
> matches the response resource.

Reviewer: Love that you included this.

Just want to tighten it up a bit.
```suggestion
**Note:** To identify the resource being operated on, the rule checks for
the request field `name`, and comparing its `(google.api.resource_reference).type`,
if present, to the response message's `(google.api.resource).type`, if present.
```
